backend/fms_core/services/experiment_run.py

```diff
@@ -88,13 +91,68 @@ def create_experiment_run(experiment_run_name,
 
     return (experiment_run, errors, warnings)
 
-def launch_experiment_run(pk):
+def start_experiment_run_processing(pk):
+    '''
+    Generates a run info file for an experiment and drops it in a spool directory
+    watched by Tech Dev, which triggers run processing to be scheduled.
+
+    If the run info file is generated without error then the run_processing_launch_date
+    timestamp is updated with the current date and time.
+
+    Args:
+        The ID of the experiment.
+
+    Returns:
+        The experiment run and a list of errors and warnings.
+    '''
     errors = []
     warnings = []
 
-    # For now, just set the launch timestamp on the experiment run...
-    experiment_run = ExperimentRun.objects.get(id=pk)
-    experiment_run.run_processing_launch_date = datetime.now()
-    experiment_run.save()
-   
+    experiment_run = None
+    try:
+        experiment_run = ExperimentRun.objects.get(id=pk)
+    except ExperimentRun.DoesNotExist:
+        errors.append(f'Experiment run with id {pk} not found.')
+
+    if experiment_run is not None:
+        try:
+            generate_run_info_file(experiment_run)
+
+            experiment_run.run_processing_launch_date = timezone.now()
+            experiment_run.save()
+        except Exception as e:
+            errors.append(str(e))
+               
     return (experiment_run, errors, warnings)
+
+def get_run_info_for_experiment(pk):
```

Comment:
This service hardly does anything ... it is a near empty wrapper to generate_run_info. I would get rid of it.

Reply:
Personally, I think it's cleaner if the viewset only uses services from the `experiment_run` services file, rather than calling functions directly in `experiment_run_info`. This function also does exception handling to provide a function that returns errors and warnings, to conform with all of the other service calls. I think it's worth keeping this "service" function, even if it doesn't do a lot.

Reply:
It does have some structural and functional potential. It is however the single call of another placeholder function. The real function is in the experiment_run_info services. if the placement of code is the only reason to create a function you can move experiment_run_info services into experiment_run services we'll save a few dozen lines of code ;). The experiment_run_info is not really a service. It is because we don't have a better place for it but it does not wrap any model (best approximation is experiment_run). You can leave it if you really want but it really lessen code readability. you have 4 functions with nearly the same name that do pretty much the same thing (which is next to nothing). All this error checking can be moved along. I could see a functional use for start_experiment_run_processing but this one is pretty much just a redirection.